core/src/test/java/io/confluent/rest/metrics/Jetty429MetricsDosFilterListenerIntegrationTest.java

```diff
@@ -0,0 +1,299 @@
+import static org.junit.jupiter.api.Assertions.fail;
+
+import io.confluent.rest.Application;
+import io.confluent.rest.RestConfig;
+import io.confluent.rest.TestMetricsReporter;
+import io.confluent.rest.TestRestConfig;
+import java.net.URI;
+import java.time.Duration;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.core.Configurable;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import org.apache.kafka.common.metrics.KafkaMetric;
+import org.eclipse.jetty.server.Server;
+import org.glassfish.jersey.server.ServerProperties;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+@Tag("IntegrationTest")
+class Jetty429MetricsDosFilterListenerIntegrationTest {
+
+  private static final int DOS_FILTER_MAX_REQUESTS_PER_CONNECTION_PER_SEC = 25;
+
+  private ScheduledExecutorService executor;
+  private Server server;
+  private Client client;
+
+  @BeforeEach
+  public void setUp() throws Exception {
+    TestMetricsReporter.reset();
+
+    Properties props = new Properties();
+    props.setProperty("debug", "false");
+    props.put(RestConfig.METRICS_REPORTER_CLASSES_CONFIG, "io.confluent.rest.TestMetricsReporter");
+    // enabled dos filters
+    props.put("dos.filter.enabled", true);
+    props.put("dos.filter.delay.ms", -1L); // to reject request, i.e 429
+    props.put("dos.filter.max.requests.per.connection.per.sec",
+        DOS_FILTER_MAX_REQUESTS_PER_CONNECTION_PER_SEC); // local dos filter
+    // NOTE: we can't test both global and local dos filter in integration tests because
+    // all requests are sent from only one client ip, the only dos filter in effect in the tests
+    // is local dos filer
+    props.put("dos.filter.max.requests.per.sec", 100); // global dos filter
+
+    TestRestConfig config = new TestRestConfig(props);
+    ApplicationWithDoSFilterEnabled app = new ApplicationWithDoSFilterEnabled(config);
+    server = app.createServer();
+    server.start();
+
+    executor = Executors.newScheduledThreadPool(4);
+    client = ClientBuilder.newClient(app.resourceConfig.getConfiguration());
+  }
+
+  @AfterEach
+  public void tearDown() throws Exception {
+    server.stop();
+    server.join();
+
+    client.close();
+    awaitTerminationAfterShutdown(executor);
+  }
+
+  @Test
+  public void testDosFilterRateLimitMetrics_noRejected() {
```

Comment:
Firstly, good test!

Our current rate-limit test are disabled since they are flaky, example - 
https://github.com/confluentinc/kafka-rest/blob/1ac7e3f9a75fa35b0f511d4728e87779cfefa9e1/kafka-rest/src/test/java/io/confluent/kafkarest/ratelimit/FixedCostRateLimitFeatureTest.java#L34

Do both of these test-cases pass when you run then in a loop say 10 times?

Reply:
Made both the tests repeated 10 times